src/apis/emscripten/time.rs

```diff
@@ -41,19 +40,12 @@ pub extern "C" fn _clock_gettime(clk_id: c_int, tp: c_int, instance: &mut Instan
         tv_nsec: i32,
     }
 
-    unsafe {
-        let mut timespec = timespec {
-            tv_sec: 0,
-            tv_nsec: 0,
-        };
-        let ret = libc_clock_gettime(clk_id as _, &mut timespec);
-        if ret != 0 {
-            return ret;
-        }
+    let timespec = time::get_time();
```

Comment:
The emscripten implementation does something like:


```js
function _emscripten_get_now() { abort() }
  
  function _emscripten_get_now_is_monotonic() {
      // return whether emscripten_get_now is guaranteed monotonic; the Date.now
      // implementation is not :(
      return ENVIRONMENT_IS_NODE || (typeof dateNow !== 'undefined') ||
          ((ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) && self['performance'] && self['performance']['now']);
    }function _clock_gettime(clk_id, tp) {
      // int clock_gettime(clockid_t clk_id, struct timespec *tp);
      var now;
      if (clk_id === 0) {
        now = Date.now();
      } else if (clk_id === 1 && _emscripten_get_now_is_monotonic()) {
        now = _emscripten_get_now();
      } else {
        ___setErrNo(ERRNO_CODES.EINVAL);
        return -1;
      }
      HEAP32[((tp)>>2)]=(now/1000)|0; // seconds
      HEAP32[(((tp)+(4))>>2)]=((now % 1000)*1000*1000)|0; // nanoseconds
      return 0;
    }
```

So basically, if the `clk_id` is:
* 0 => return time::get_time();
* 1 => abort()
* (any) => setErrno (we can skip this for now), return -1.

I think we should do something similar.

Reply:
That's a good idea. 

I made a simple solution using `panic!`. If we want to handle errors differently, let me know!

Reply:
Sounds good!